:invalml
:e


2(2)
//│ ╔══[ERROR] Type error in application
//│ ║  l.5: 	2(2)
//│ ║       	^^^^
//│ ╙── because: cannot constrain  Int  <:  (Int) ->{'eff} ('app)
//│ Type: ⊥

(x => x(0))(1)
//│ ╔══[ERROR] Type error in integer literal with expected type 'x
//│ ║  l.12: 	(x => x(0))(1)
//│ ║        	            ^
//│ ╟── because: cannot constrain  Int  <:  'x
//│ ╙── because: cannot constrain  Int  <:  (Int) ->{'eff} ('app)
//│ Type: ⊥

(1).Foo#a()
//│ ╔══[ERROR] Name not found: Foo
//│ ║  l.20: 	(1).Foo#a()
//│ ╙──      	   ^^^^
//│ ╔══[ERROR] Identifier `Foo` does not name a known class symbol.
//│ ║  l.20: 	(1).Foo#a()
//│ ╙──      	   ^^^^
//│ ═══[ERROR] Not a valid class: <error>
//│ Type: ⊥


fun Oops() = 1
Oops().Oops#a()
//│ ╔══[ERROR] Identifier `Oops` does not name a known class symbol.
//│ ║  l.32: 	Oops().Oops#a()
//│ ╙──      	      ^^^^^
//│ ╔══[ERROR] Not a valid class: selection
//│ ║  l.32: 	Oops().Oops#a()
//│ ╙──      	      ^^^^^
//│ Type: ⊥

data class Oops2()
(new Oops2()).Oops2#a()
//│ ╔══[ERROR] Class 'Oops2' does not contain member 'a'.
//│ ║  l.42: 	(new Oops2()).Oops2#a()
//│ ╙──      	                    ^
//│ ╔══[ERROR] a is not a valid member in class Oops2
//│ ║  l.42: 	(new Oops2()).Oops2#a()
//│ ╙──      	     ^^^^^^^^^^^^^^^^
//│ Type: ⊥



fun inc(x) = x + 1
inc("oops")
//│ ╔══[ERROR] Type error in string literal with expected type 'x
//│ ║  l.54: 	inc("oops")
//│ ║        	    ^^^^^^
//│ ╟── because: cannot constrain  Str  <:  'x
//│ ╙── because: cannot constrain  Str  <:  Int
//│ Type: Int

fun inc(x) = x + 1
inc as Int
//│ ╔══[ERROR] Type error in reference with expected type Int
//│ ║  l.63: 	inc as Int
//│ ║        	^^^
//│ ╙── because: cannot constrain  ('x) ->{⊥} (Int)  <:  Int
//│ Type: Int


